docs/scripting-ref.rst

```diff
@@ -391,7 +393,223 @@ Compare:
+yield until the JavaScript code tells it to resume.
+
+**Signature:** ``value, error = splash:wait_for_resume(snippet, timeout)``
+
+**Parameters:**
+
+* snippet - a string with a JavaScript source code to execute. This code
+  must include a function called ``main``. The first argument to ``main``
+  is an object that has the properties ``resume`` and ``error``. ``resume``
+  is a function which can be used to resume Lua execution. It takes an optional
+  argument which will be returned to Lua in the ``value`` return value.
+  ``error`` is a function which can be called with a required string value
+  that is returned in the ``error`` return value.
+* timeout - a number which determines (in seconds) how long to allow JavaScript
+  to execute before forceably returning control to Lua. Defaults to
+  zero, which disables the timeout.
+
+**Returns:** ``value, error`` pair. When the execution is successful
+``value`` is a table that contains the JavaScript value passed to
+``splash.resume(…)``. It also contains any additional key/value pairs
+set by ``splash.set(…)``. In case of timeout or JavaScript errors ``value``
+is ``nil`` and ``error`` contains an error message string.
+
+Examples:
+
+The first, trivial example shows how to transfer control of execution from Lua
+to JavaScript and then back to Lua. This command will tell JavaScript to
+sleep for 3 seconds and then return to Lua. Note that this is an async
+operation: the Lua event loop and the JavaScript event loop continue to run
+during this 3 second pause, but Lua will not continue executing the current
+function until JavaScript calls ``splash.resume()``.
+
+.. code-block:: lua
+
+    function main(splash)
+
+        local value, error = splash:wait_for_resume([[
+            function main(splash) {
+                setTimeout(function () {
+                    splash.resume();
+                }, 3000);
+            }
+        ]])
+
+        -- value is {}
+        -- error is nil
+
+    end
+
+``value`` is set to an empty table to indicate that nothing was returned
+from ``splash.resume``. You can use ``assert(splash:wait_for_resume(…))``
+even when JavaScript does not return a value because the empty table signifies
+success to ``assert()``.
+
+.. note::
+
+    The ``main()`` function is required, and you will get an error if
+    you do not include it. The first argument to this function can have any
+    name you choose, of course. We will call it ``splash`` by convention in
+    this documentation.
+
+The next example shows how to return a value from JavaScript to Lua.
+You can return booleans, numbers, strings, arrays, or objects.
+
+.. code-block:: lua
+
+    function main(splash)
+
+        local value, error = splash:wait_for_resume([[
+            function main(splash) {
+                setTimeout(function () {
+                    splash.resume([1, 2, 'red', 'blue']);
+                }, 3000);
+            }
+        ]])
+
+        -- value is {return=[1, 2, 'red', 'blue']}
```

Comment:
'return' is a reserved word in lua, and this causes inconveniences - e.g. you can't write `value.return` to get the value.